#!/usr/bin/env python
PACKAGE = "hw07_crist"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('enable',         bool_t,  0, 'Enable',        False)
gen.add('red_offset',   double_t,  0, 'Red Offset',   1.0, -1.0,   1.0)
gen.add('red_gain',     double_t,  0, 'Red Gain',     -1.0, -2.0,   2.0)
gen.add('blue_offset',  double_t,  0, 'Blue Offset',  1.0, -1.0,   1.0)
gen.add('blue_gain',    double_t,  0, 'Blue Gain',    1.0, -2.0,   2.0)
gen.add('green_offset', double_t,  0, 'Green Offset', 0.0, -1.0,   1.0)
gen.add('green_gain',   double_t,  0, 'Green Gain',   2.0, -2.0,   2.0)

exit(gen.generate(PACKAGE, "hw07_crist", "SpeedSettingsDynCfg"))


